runtime/src/bank/builtins/core_bpf_migration/mod.rs

```diff
@@ -311,44 +282,24 @@ mod tests {
     const TEST_ELF: &[u8] =
         include_bytes!("../../../../../programs/bpf_loader/test_elfs/out/noop_aligned.so");
 
-    const PROGRAM_DATA_OFFSET: usize = UpgradeableLoaderState::size_of_programdata_metadata();
```

Comment:
What was wrong with `PROGRAM_DATA_OFFSET`?

Reply:
Isn't that the size if the upgrade authority is `Some`?

Reply:
Yeah, those size helpers only work for `Some` values. I've included code to handle `None` in here, in case we want to go with `None` for authority.

Reply:
bincode might truncate the trailing bytes of None, but an Option always has the same size. The loader does always place the ELF at `size_of_programdata_metadata()` and I think this should be consistent here. Also, can the authority even be None for the core program migration?

Reply:
Technically the authority really shouldn't be, but there's nothing in the SIMDs so far that says it can't be frozen.